client/landing/stepper/declarative-flow/free-post-setup.ts

```diff
@@ -0,0 +1,49 @@
+import { FREE_POST_SETUP_FLOW } from '@automattic/onboarding';
+import { translate } from 'i18n-calypso';
+import { useSiteSlug } from '../hooks/use-site-slug';
+import { recordSubmitStep } from './internals/analytics/record-submit-step';
+import FreePostSetup from './internals/steps-repository/free-post-setup';
+import { ProvidedDependencies } from './internals/types';
+import type { Flow } from './internals/types';
+
+const freePostSetup: Flow = {
```

Comment:
Is there a consolidated place we can read about the reasoning behind "Post setup flows"? 

I think documentation will be important since this implementation appears be an established convention for how we'll deal with the common use case of needing to return to previous steps in a flow ( after site creation )

If not, can we consider adding notes somewhere?

Reply:
We don't have documentation on why we do post-setup flows anywhere. The reason is that we are trying to avoid backward navigation within a stepper flow. This allows each stepper step/screen to be simple and avoid having to accommodate different conditions when accessed at different times in the flow. 

Since we can't go back to the setup screen (or others) we spin up mini-flows to allow the equivalent.

For the origin of the convo about this see [this comment](https://github.com/Automattic/wp-calypso/pull/67523#pullrequestreview-1107102774) on a past (closed). Also, telated to this, we recently [deprecated some of the Stepper navigation methods](https://github.com/Automattic/wp-calypso/pull/68172) that allow navigation between steps within a flow. 

If it is helpful, I can write up a P2 on this.